Glue Code for Discussion

ben-jones opened the pull request:
Everyone,

Please note that this is not working code. I have generated pseudocode for the glue code which will incorporate everything that we have done. Please look over the code and provide feedback since the only purpose of this pull request is to generate discussion.

Notes: conn is just a frame object

ben-jones committed:
Added initial htpt code

Added initial version of glue code for the project

ben-jones committed:
Added initial htpt code

Added initial version of glue code for the project

ben-jones committed:
Working version of glue code

Created a version of the glue code to generate discussion.

ben-jones committed:
Working version of glue code

Created a version of the glue code to generate discussion.

gsathya commented on htpt/htpt.py:
```diff
@@ -0,0 +1,104 @@
+# imports
+from datetime import datetime
+# local imports
+from htpt import frame.py
```
`from htpt import frame`

ben-jones committed:
Fixed typo

Had a typo where I was using url encoding from the server to the
client. I fixed it to use image encoding

ben-jones committed:
Fixed typo

Had a typo where I was using url encoding from the server to the
client. I fixed it to use image encoding

gsathya commented on htpt/htpt.py:
```diff
@@ -0,0 +1,104 @@
+  conns.append(frame.initConn())
+  conns[0].connect(torBridgeAddress, torBridgePassword)
+  conns[0].timeout = datetime.now()
+  while 1:
```
this should be async stuff. if we get data from both Tor and from the internet, then one of them might get dropped

gsathya replied:
Also, this while loop is going to waste cpu cycles. We should use an event loop (twisted or gevent)

ben-jones committed:
Initial client code

Made significant updates to glue code so that the client code is
basically done. All that is left for the client is updating the
headless webkit lines so that they actually do something

ben-jones committed:
Initial client code

Made significant updates to glue code so that the client code is
basically done. All that is left for the client is updating the
headless webkit lines so that they actually do something

ben-jones committed:
Updates to server code

Right now, the server code cannot be finished, so I am commiting
what I do have done for comments. There are 2 main issues
1) the sequence num and session id code needs to be refactored.
It really needs an initialize function and we need to make sure
that the encode/decode functions and the framing stuff can play
together nicely
2) we need a way to bundle the assembler, disassembler, and Internet
connections together. Any advice on abstractions is welcome.
Otherwise, will probably default to a data structure which stores
the session ID and each parameter

ben-jones committed:
Updates to server code

Right now, the server code cannot be finished, so I am commiting
what I do have done for comments. There are 2 main issues
1) the sequence num and session id code needs to be refactored.
It really needs an initialize function and we need to make sure
that the encode/decode functions and the framing stuff can play
together nicely
2) we need a way to bundle the assembler, disassembler, and Internet
connections together. Any advice on abstractions is welcome.
Otherwise, will probably default to a data structure which stores
the session ID and each parameter

ben-jones committed:
Added better interactions with framing code

Updated htpt.py to use the correct framing functions and used the
updated api for intializing connections from clients. Also, started
adding elements of url parsing with flask.

ben-jones committed:
Added basic flask code

Added basic function implementation using flask to get data from
apache

ben-jones committed:
Refactored to reduce unnecessary code

Significantly reduced the amount of code in htpt.py since most of
it was redundant

ben-jones committed:
Merge branch 'htpt' of github.com:ben-jones/htpt into htpt

Conflicts:
	htpt/htpt.py

ben-jones committed:
First working version!!!!!!!!!!!!!

I think git may have lost my commit history, and if so, then this
is a number of changes which integrates the functionality of htpt
and fixed bugs

ben-jones committed:
Fixed file name
